### ISSUE TITLE
[BUG] API error when calling log_params with a dict containing an empty string as value

### ISSUE THREAD

**Issue by mautini** [mlflow/mlflow#6253]
### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

1.27.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux / MacOS
- **Python version**: 3.10.5
- **yarn version, if running the dev UI**:

### Describe the problem

Using the last version of mlflow (and Python client), when I try to log params in a run using the log_params method (and a dict in argument), I got an API error if a value of my dict is an empty string.

The following code works : `mlflow.log_param('test', '')`
But this one not : `mlflow.log_params({'test': ''})`

### Tracking information

MLflow version: 1.27.0

tracking uri and artifact uri are sensitive informations

### Code to reproduce issue

```python
mlflow.log_params({'test': ''})
```

### Other info / logs

```python-traceback
{RestException}INVALID_PARAMETER_VALUE: Invalid value [{'key': 'test', 'value': ''}] for parameter 'params' supplied. Hint: Value was of type 'list'. See the API docs for more information about request parameters.
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@mautini https://github.com/mlflow/mlflow/pull/6179 fixed this issue :)

**Comment by britishvnr** [comment]
Trying with 1.27.0, it apparently wasn't fixed everywhere:   File "/workspace/entrypoints/../src/train.py", line 236, in <module>
    main(classifier_training_args, data_training_args, model_args)
  File "/workspace/entrypoints/../src/train.py", line 65, in main
    mlflow.log_params(params=preprocess_args)
  File "/workspace/env/lib/python3.9/site-packages/mlflow/tracking/fluent.py", line 675, in log_params
    MlflowClient().log_batch(run_id=run_id, metrics=[], params=params_arr, tags=[])
  File "/workspace/env/lib/python3.9/site-packages/mlflow/tracking/client.py", line 918, in log_batch
    self._tracking_client.log_batch(run_id, metrics, params, tags)
  File "/workspace/env/lib/python3.9/site-packages/mlflow/tracking/_tracking_service/client.py", line 315, in log_batch
    self.store.log_batch(
  File "/workspace/env/lib/python3.9/site-packages/mlflow/store/tracking/rest_store.py", line 309, in log_batch
    self._call_endpoint(LogBatch, req_body)
  File "/workspace/env/lib/python3.9/site-packages/mlflow/store/tracking/rest_store.py", line 56, in _call_endpoint
    return call_endpoint(self.get_host_creds(), endpoint, method, json_body, response_proto)
  File "/workspace/env/lib/python3.9/site-packages/mlflow/utils/rest_utils.py", line 256, in call_endpoint
    response = verify_rest_response(response, endpoint)
  File "/workspace/env/lib/python3.9/site-packages/mlflow/utils/rest_utils.py", line 185, in verify_rest_response
    raise RestException(json.loads(response.text))
mlflow.exceptions.RestException: INVALID_PARAMETER_VALUE: Invalid value [{'key': 'preprocess.input_path', 'value': 'input_data/theme/top_50_theme_8'}, {'key': 'preprocess.output_path', 'value': 'data'}, {'key': 'preprocess.threshold', 'value': '30'}, {'key': 'preprocess.head', 'value': '0'}, {'key': 'preprocess.filter', 'value': ''}, {'key': 'preprocess.chunk', 'value': '0'}, {'key': 'preprocess.max_depth', 'value': '1'}, {'key': 'preprocess.partition_factor', 'value': '1'}, {'key': 'preprocess.text', 'value': 'both'}, {'key': 'preprocess.n_threads', 'value': '0'}, {'key': 'preprocess.no_dropdup', 'value': 'True'}] for parameter 'params' supplied. Hint: Value was of type 'list'. See the API docs for more information about request parameters.

**Comment by harupy** [comment]
@britishvnr The patch will be shipped in 1.28.0.
